### PR TITLE
Config Descriptions

### LINKED ISSUE #15: [Feature Request]: Config Descriptions
Wakes is an amazing mod. It works really well, and looks fantastic. However, I recently found myself playing with the config, and had no idea what option did what. YetAnotherConfigLib has an amazing way to easily add descriptions. They also aren't like normal Minecraft tooltips, which means you can have much longer descriptions without taking up all of the screenspace. On top of that, YACL also allows for image descriptions on-top of text descriptions! However, the downside to that is, depending on how many images you have, the file size will noticeably increase. So images must be used wisely.

I'm working on a PR to add text descriptions if you haven't already added them locally.

### PR DESCRIPTION
**Pull request by Superkat32**
Fixes #15 

The only thing that has been added is config text descriptions. The formatting of the en_us.json file has been done similar to how you already have it. The only options which don't have config descriptions are those in the debug category. I also changed the name of the "Velocity required to produce a wake" to just "Velocity required", having its description include the "to produce a wake" instead.

Beyond the config descriptions, Wakes is an amazing mod and really fun to use! I love it so much!

### CODE REVIEW

**Comment by Goby56** [comment]
Yeah descriptions would be really handy. I have not yet begun adding them so this PR would be really helpful. I'll take a look at it tomorrow.

Thanks for your contribution and glad you are enjoying the mod!

File: src/main/resources/assets/wakes/lang/en_us.json [hunk 1]
```diff
@@ -24,14 +29,22 @@
   "wake_spawn_rule.wakes.only_wakes": "Only wakes",
   "wake_spawn_rule.wakes.only_splashes": "Only splashes",
   "wake_spawn_rule.wakes.disabled": "Disabled",
+  "description.wakes.spawning_rules": "Defines the wakes and splashes spawning rules.",
   "option.wakes.wakes_in_running_water": "Wakes in running water",
+  "description.wakes.wakes_in_running_water": "Should wakes spawn in running water?",
 
   "option.wakes.wave_speed": "Wave speed",
+  "description.wakes.wave_speed": "The speed which the wakes travel and despawn at.",
```
Comment (Goby56) [inline src/main/resources/assets/wakes/lang/en_us.json:37]: While I do state this is wake speed I realise now it's quite a bad name for it. Technically speaking and in terms of the wave simulation it is speed but when changing the value the wave doesn't really propagate faster, only farther. I might change this to something like "Wave propagation factor" or something similar.

File: src/main/resources/assets/wakes/lang/en_us.json [hunk 2]
```diff
@@ -24,14 +29,22 @@
...
   "option.wakes.initial_wave_strength": "Initial wave strength",
+  "description.wakes.initial_wave_strength": "The distance the wakes will travel.",
```
Comment (Goby56) [inline src/main/resources/assets/wakes/lang/en_us.json:39]: Minor thing here: This option might result in wakes traveling farther but that may just be a result of a higher starting value which means it takes longer for the wave to die down. The description should state that it's primarily a strength metric that in turn cause the wakes to travel longer distances.

**Review by Goby56 — APPROVED** [review]
Good descriptions overall. Just some things I wanted to highlight that I may change in the future because of my poor inital wording of some of the options.

File: src/main/resources/assets/wakes/lang/en_us.json (hunk 1 above)
Comment (Superkat32) [reply src/main/resources/assets/wakes/lang/en_us.json:37]: Pretty sure this is one of the options I struggled with making a description for. It seemed like it did a lot of things at once. I made the descriptions based on my testing of changing the settings in-game, with one or two having some code reading alongside it.

In your defense, naming config options is surprisingly difficult.

**Pull request merged**